overlord/snapstate/backend/setup.go

```diff
@@ -57,6 +57,13 @@ func (b Backend) SetupSnap(snapFilePath, instanceName string, sideInfo *snap.Sid
 		return snapType, err
 	}
 
+	if s.InstanceKey != "" {
+		err := os.MkdirAll(snap.BaseDir(s.SnapName()), 0755)
```

Comment:
do we need to remove this in the defer above in some case?

Reply:
This has to be cleaned up by the caller in snapstate. Here we do not know if the directory is perhaps used by another snap once we get to the cleanup part.

Reply:
you mean is covered by the new code in doMountSnap error handling?